lib/spack/spack/spec.py

```diff
@@ -738,24 +774,22 @@ def __init__(self, spec):
         super(FlagMap, self).__init__()
         self.spec = spec
 
-    def satisfies(self, other, strict=False):
-        if strict or (self.spec and self.spec._concrete):
-            return all(f in self and set(self[f]) == set(other[f]) for f in other)
-        else:
+    def satisfies(self, other):
+        return all(f in self and set(self[f]) == set(other[f]) for f in other)
```

Comment:
Why set? Flags are order dependent, `-ffeature -fno-feature` != `-fno-feature -ffeature`.

I think I would just stick to a interpretation of *flags being a variant with string value type. So, for satisfies you just drop `set` and for intersects you stick to the flags set in either an require equal values.

Reply:
Flags semantic has been changed in 3a5b534fa9c8cb42e71fe715ea9859a8a18d5528